docs/entities/renderer.md

```diff
@@ -0,0 +1,226 @@
+        );
+        // We can now add children to any PartDefinition, thus creating a hierarchy.
+        PartDefinition part1 = root.addOrReplaceChild(...);
+        PartDefinition part2 = head.addOrReplaceChild(...);
+        PartDefinition part3 = part1.addOrReplaceChild(...);
+        // At the end, we create a LayerDefinition from the MeshDefinition.
+        // The two integers are the expected dimensions of the texture; 64x32 in our example.
+        return LayerDefinition.create(mesh, 64, 32);
+    }
+}
+```
+
+Note that in the above example, we directly extend `EntityModel`; depending on your use case, it might be more appropriate to use one of the subclasses instead. When creating a new model, it is recommended you have a look at whatever existing model is closest to your use case, and then work from there.
+
+:::tip
+The [Blockbench][blockbench] modeling program is a great help in creating entity models. To do so, choose the Modded Entity option when creating your model in Blockbench.
+
+Blockbench also has an option to export models as a `LayerDefinition` creation method, which can be found under `File -> Export -> Export Java Entity`.
+:::
+
+### Registering a Layer Definition
+
+Once we have our entity layer definition, we also need to register it in `EntityRenderersEvent.RegisterLayerDefinitions`. To do so, we need a `ModelLayerLocation`, which essentially acts as an identifier for our layer (remember, one entity can have multiple layers).
+
+```java
+// Our ModelLayerLocation.
+public static final ModelLayerLocation MY_LAYER = new ModelLayerLocation(
+    // Should be the name of the entity this layer belongs to.
+    // May be more generic if this layer can be used on multiple entities.
+    ResourceLocation.fromNamespaceAndPath("examplemod", "example_entity"),
+    // The name of the layer itself.
+    "main"
+);
+
+@SubscribeEvent
+public static void registerLayerDefinitions(EntityRenderersEvent.RegisterLayerDefinitions event) {
+    // Add our layer here.
+    event.add(MY_LAYER, MyEntityModel::createBodyLayer);
+}
+```
+
+### Adding a Layer Definition to an Entity
+
+In some contexts, you might also want to add a new layer to an existing entity. For example, you might want to render some extra equipment on an entity when worn. This can be done like so:
+
+```java
+@SubscribeEvent
+public static void addLayers(EntityRenderersEvent.AddLayers event) {
+    // Add a layer to every single entity type.
+    for (EntityType<?> entityType : event.getEntityTypes()) {
+        // Get our renderer.
+        EntityRenderer<?, ?> renderer = event.getRenderer(entityType);
+        // Null-check the renderer. You could add more checks here. For example, a common check is
+        // `instanceof LivingEntityRenderer<?, ?>` to only target living entity renderers.
+        if (renderer != null) {
+            // Add the layer to the renderer. Reuses the ModelLayerLocation from above.
+            renderer.addLayer(MY_LAYER);
+        }
```

Comment:
This is wrong in two ways:
- This event has nothing to do with `LayerDefinition`s, it's related to `RenderLayer`s. A `RenderLayer` however can indeed render an `EntityModel` created from a `LayerDefinition` after the latter is baked
- Only renderers which implement `RenderLayerParent` can have `RenderLayer`s added to them. In vanilla this only applies to `LivingEntityRenderer` and its subclasses

`RenderLayer`s should be dealt with in a separate section. This section should instead explain how to go from a `LayerDefinition` to the custom `EntityModel` in the renderer's constructor.